src/bundle/Controller/LocationController.php

```diff
@@ -81,4 +99,38 @@ public function updateDefaultSortOrderAction(
 
         return $this->reloadTab('details', $content->id, $location->id);
     }
+
+    /**
+     * Delete locations based on form submit.
+     *
+     * @param FormInterface $form
+     * @param mixed $redirectLocationId
+     *
+     * @return bool
+     *  Whether to reset location or not.
+     */
+    private function deleteLocationsBasedOnFormSubmit(FormInterface $form, $redirectLocationId)
+    {
+        $locationIds = array_keys($form->get('removeLocations')->getData());
+
+        if ($form->get('delete')->isClicked()) {
+            $this->uiLocationService->deleteLocations($locationIds);
+
+            if (in_array($redirectLocationId, $locationIds)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private function addLocationBasedOnFormSubmit(FormInterface $form, Content $content)
+    {
+        if ($form->get('add')->isClicked()) {
```

Comment:
About this test, and the equivalent one in the `deleteLocations` method above, don't you think that it would make more sense to test for that in `actionsAction` ?